pyproject.toml

```diff
@@ -29,6 +29,7 @@ dependencies = [
     'numcodecs>=0.10.0',
     'fsspec>2024',
     'crc32c',
+    'universal-pathlib',
```

Comment:
Should we make upath an optional dependency?

Reply:
Yes please!

Reply:
👍 done